### ISSUE TITLE
Current HEAD of master branch crashes on building sphinx-doc itself

### ISSUE THREAD

**Issue by tk0miya** [sphinx-doc/sphinx#3350]
### Problem
- sphinx-build crashes on building the document of sphinx itself

#### Procedure to reproduce the problem
```
tox -e py27
```
or
```
sphinx-build doc html
```

#### Error logs / results
```
# Sphinx version: 1.6
# Python version: 2.7.13 (CPython)
# Docutils version: 0.13.1 release
# Jinja2 version: 2.9.4
# Last messages:

# Loaded extensions:
#   sphinx.ext.viewcode (1.6) from /Users/tkomiya/work/sphinx/.tox/py27/lib/python2.7/site-packages/sphinx/ext/viewcode.pyc
#   sphinx.ext.todo (1.6) from /Users/tkomiya/work/sphinx/.tox/py27/lib/python2.7/site-packages/sphinx/ext/todo.pyc
#   sphinx.ext.autodoc (1.6) from /Users/tkomiya/work/sphinx/.tox/py27/lib/python2.7/site-packages/sphinx/ext/autodoc.pyc
#   sphinx.ext.doctest (1.6) from /Users/tkomiya/work/sphinx/.tox/py27/lib/python2.7/site-packages/sphinx/ext/doctest.pyc
#   alabaster (0.7.9) from /Users/tkomiya/work/sphinx/.tox/py27/lib/python2.7/site-packages/alabaster/__init__.pyc
#   sphinx.ext.extlinks (1.6) from /Users/tkomiya/work/sphinx/.tox/py27/lib/python2.7/site-packages/sphinx/ext/extlinks.pyc
#   sphinx.ext.autosummary (1.6) from /Users/tkomiya/work/sphinx/.tox/py27/lib/python2.7/site-packages/sphinx/ext/autosummary/__init__.pyc
Traceback (most recent call last):
  File "/Users/tkomiya/work/sphinx/.tox/py27/lib/python2.7/site-packages/sphinx/cmdline.py", line 305, in main
    app.build(opts.force_all, filenames)
  File "/Users/tkomiya/work/sphinx/.tox/py27/lib/python2.7/site-packages/sphinx/application.py", line 348, in build
    self.builder.build_update()
  File "/Users/tkomiya/work/sphinx/.tox/py27/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 277, in build_update
    'out of date' % len(to_build))
  File "/Users/tkomiya/work/sphinx/.tox/py27/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 345, in build
    self.write(docnames, list(updated_docnames), method)
  File "/Users/tkomiya/work/sphinx/.tox/py27/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 382, in write
    self._write_serial(sorted(docnames))
  File "/Users/tkomiya/work/sphinx/.tox/py27/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 389, in _write_serial
    doctree = self.env.get_and_resolve_doctree(docname, self)
  File "/Users/tkomiya/work/sphinx/.tox/py27/lib/python2.7/site-packages/sphinx/environment/__init__.py", line 1088, in get_and_resolve_doctree
    self.resolve_references(doctree, docname, builder)
  File "/Users/tkomiya/work/sphinx/.tox/py27/lib/python2.7/site-packages/sphinx/environment/__init__.py", line 1139, in resolve_references
    typ, target, node, contnode)
  File "/Users/tkomiya/work/sphinx/.tox/py27/lib/python2.7/site-packages/sphinx/domains/cpp.py", line 4983, in resolve_xref
    target, node, contnode)[0]
  File "/Users/tkomiya/work/sphinx/.tox/py27/lib/python2.7/site-packages/sphinx/domains/cpp.py", line 4960, in _resolve_xref_inner
    if not node['refexplicit'] and declaration.objectType == 'function':
  File "/Users/tkomiya/work/sphinx/.tox/py27/lib/python2.7/site-packages/docutils/nodes.py", line 567, in __getitem__
    return self.attributes[key]
KeyError: 'refexplicit'
```

#### Expected results
- build succeeded with no errors.

### Reproducible project / your project
- sphinx-doc/sphinx

### Environment info
- OS: Mac
- Python version: 2.7
- Sphinx version: 1.6

**Comment by jfbu** [comment]
hi @tk0miya this was reported at https://github.com/sphinx-doc/sphinx/issues/2952#issuecomment-272709980 and it has been fixed in stable already at  1498897 can you merge stable ?

**Comment by tk0miya** [comment]
Thanks! I just merged.
